src/backend/cuda/platform.cpp

```diff
@@ -203,6 +203,12 @@ bool isDoubleSupported(int device) {
     return true;
 }
 
+bool isHalfSupported(int device) {
+    auto prop = getDeviceProp(device);
+    float compute = prop.major * 1000 + prop.minor * 10;
+    return compute >= 5030;
```

Comment:
hum does it mean that any device setup with CudaComp 5.3+ will really support half ?

Reply:
Yes, only those devices have support for half operations. I can try to incorporate older devices but that will take more time.